fpdf/html.py

```diff
@@ -280,8 +283,25 @@ def handle_data(self, data):
                 align=self.align[0].upper(),
                 link=self.href,
             )
+        elif self.pre_formatted: #for pre blocks
+            self.pdf.write(self.h, data)
+
+        elif self.follows_fmt_tag: #don't trim leading whitespace if following a format tag
+            data = data.replace("\n", " ") #flatten multiline strings
```

Comment:
> The "\n" special sequence is included in "\s". Why do you consider it necessary to handle it seperately like this?

True, just excess caution I think! I didn't want to break existing functionality but \s does indeed do the same task, I'll remove the \n sequence.

> There's another pitfall though: In Unicode mode, the non-breaking space characters ('\u00a0" NO-BREAK SPACE and "\u202f" NARROW NO-BREAK SPACE, possibly others) are also included in "\s", but should actually be preserved verbatim.

I was not aware, thanks for catching this. I'll add these cases to the test as well. 

The small catch with the inverse RE is that it then pads the nbsp characters with another whitespace character, so they will need to be handled differently. 

Just pushed a fix that passes the modified test. Passing a function to `re.sub() ` allows for some fine tuning of how the `\s` matches are handled. If any other problematic whitespace chars are discovered in the future they can be added to the `leading_whitespace_repl()` and `whitespace_repl()` functions, in theory.

Reply:
> The small catch with the inverse RE is that it then pads the nbsp characters with another whitespace character, so they will need to be handled differently.

If there is no other whitespace next to a nbsp, then none will get added to it. If my memory doesn't fail me completely, then that is also how the web browsers handle it.

```python
>>> t = ' ab\u00a0c def \u00a0ghi'
>>> wordpat = re.compile('[\u00a0\u202f\S]+')
>>> t = ' ab\u00a0c def \u00a0ghi '
>>> resl = wordpat .findall(t)
>>> resl
['ab\xa0c', 'def', '\xa0ghi']
>>> ' '.join(resl)
'ab\xa0c def \xa0ghi'
```

Reply:
Ah you're right, the padding I was seeing was a user error on my end in trying to generate a correct pdf for testing. My mistake.
 
Related to that though, the other catch I was finding was in leading/trailing whitespace around format tags. Ex. `<i>italic</i> <b>Bold</b> text` -> _italic_**bold**text. Since they each trigger a new data block. 

The new commit extending re.sub() seems to fix all of these cases, but I can explore addressing them with the inverse RE if that approach is preferred, just let me know.

Reply:
> `<i>italic</i> <b>Bold</b> text` -> _italic_**bold**text.

Yes, that's why you need to check for whitespace at the beginning and end of a data fragment seperately.
An additional special case are data fragments consisting of only whitespace, like `</i> <b>` in your example.

Also, in the original issue description you'll find it mentioned that it is necessary to remember whether the last data fragment ended with whitespace, because then you have to skip any initial whitespace of the next one.